ergo-lib/src/wallet/multi_sig.rs

```diff
@@ -349,7 +354,10 @@ pub fn extract_hints(
             real_secrets_to_extract.as_slice(),
             simulated_secrets_to_extract.as_slice(),
             proof.as_slice(),
-        );
+        )
+        .map_err(|_e| {
+            TxSigningError::ProverError(Unexpected("Unexpected Sigma tree Value".to_string()), i)
+        });
```

Comment:
```suggestion
        })?;
        hints_bag.add_hints_for_input(i, bag);
```
and you don't need the following `match` which silences the error case anyway.

Reply:
Good Point! i changed it.